spacewalk/certs-tools/mgr_ssl_cert_setup.py

```diff
@@ -427,40 +425,44 @@ def generateCertWithChainFile(serverCert, workdir, certData):
     return retContent
 
 
-def deployApache(workdir):
+def deployApache(apache_cert_content, server_key_content):
     if os.path.exists(APACHE_KEY_FILE):
         os.remove(APACHE_KEY_FILE)
     if os.path.exists(APACHE_CRT_FILE):
         os.remove(APACHE_CRT_FILE)
-    shutil.copy(os.path.join(workdir, SRV_KEY_NAME), APACHE_KEY_FILE)
-    if os.path.exists(os.path.join(workdir, APACHE_CRT_NAME)):
-        shutil.copy(os.path.join(workdir, APACHE_CRT_NAME), APACHE_CRT_FILE)
+    with open(APACHE_KEY_FILE, "w") as f:
+        f.write(server_key_content)
+    if apache_cert_content:
+        with open(APACHE_CRT_FILE, "w") as f:
+            f.write(apache_cert_content)
     # exists on server and proxy
     os.system("/usr/bin/spacewalk-setup-httpd")
 
 
-def deployJabberd(workdir):
+def deployJabberd(jabber_cert_content):
     j_uid, j_gid = getUidGid("jabber", "jabber")
     if j_uid and j_gid:
         if os.path.exists(JABBER_CRT_FILE):
             os.remove(JABBER_CRT_FILE)
-        if os.path.exists(os.path.join(workdir, JABBER_CRT_NAME)):
-            shutil.copy(os.path.join(workdir, JABBER_CRT_NAME), JABBER_CRT_FILE)
+        if jabber_cert_content:
```

Comment:
Also this check is obsolete as we test the content before.
But we should add the error message to `_main()` when we abort there.